string location
bool violation_detected
float32 timestamp
